Bottom: ed67cd296b290be851a24d70bd2fd8558d0fbe98
Top:    2f8d63dfb3f73624a867929b3b6526f87210d460
Author: Andrew Cooper <andrew.cooper@novatechweb.com>
Date:   2018-08-16 09:17:44 -0500

Enable change hooks from Gitlab


---
diff --git a/ansible-playbook/group_vars/all/credentials.yml b/ansible-playbook/group_vars/all/credentials.yml
index 69052558..888b3ffb 100644
--- a/ansible-playbook/group_vars/all/credentials.yml
+++ b/ansible-playbook/group_vars/all/credentials.yml
@@ -3,6 +3,7 @@
 buildbot_admin_password:             '{{ lookup("password", playbook_dir + "/credentials/buildbot/admin"                      , length=20) }}'
 buildbot_worker_ntel_password:       '{{ lookup("password", playbook_dir + "/credentials/buildbot/worker_ntel"                , length=20) }}'
 buildbot_worker_ptxdist_password:    '{{ lookup("password", playbook_dir + "/credentials/buildbot/worker_ptxdist"             , length=20) }}'
+buildbot_webhook_secret:             '{{ lookup("password", playbook_dir + "/credentials/buildbot/webhook"                    , length=20) }}'
 buildsystem_database_build_password: '{{ lookup("password", playbook_dir + "/credentials/buildsystem/database/readuser") }}'
 buildsystem_database_root_password:  '{{ lookup("password", playbook_dir + "/credentials/buildsystem/database/root"           , length=20) }}'
 buildsystem_ssh_prod_user_password:  '{{ lookup("password", playbook_dir + "/credentials/buildsystem/ssh/prod"                , length=20) }}'
diff --git a/roles/buildbot-master/defaults/main.yml b/roles/buildbot-master/defaults/main.yml
index ee40c325..32cde229 100644
--- a/roles/buildbot-master/defaults/main.yml
+++ b/roles/buildbot-master/defaults/main.yml
@@ -48,6 +48,7 @@ container_env:
   GITLAB_URI: 'https://{{ gitlab_hostname }}/'
   GITLAB_OAUTH_ID: '{{ gitlab_buildbot_auth_id }}'
   GITLAB_OAUTH_SECRET: '{{ gitlab_buildbot_auth_secret }}'
+  GITLAB_WEBHOOK_SECRET: '{{ buildbot_webhook_secret }}'
   LDAP_URI: 'ldap://{{ openldap_hostname }}:389/'
   LDAP_BIND_USER: 'cn=proxyagent,dc=novatech'
   LDAP_BIND_PW: '{{ openldap_proxyagent_password }}'
diff --git a/roles/buildbot-master/files/master.cfg b/roles/buildbot-master/files/master.cfg
index c453a09b..a12bdaa8 100644
--- a/roles/buildbot-master/files/master.cfg
+++ b/roles/buildbot-master/files/master.cfg
@@ -94,6 +94,11 @@ c['www'] = {
                 ]),
         ],
     ),
+    'change_hook_dialects': {
+        'gitlab': {
+            'secret': os.getenv('GITLAB_WEBHOOK_SECRET')
+        }
+    }
 }
 
 # PROJECT IDENTITY
